commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt

```diff
@@ -49,28 +47,28 @@ internal class TsvChunkGenerator(private val parser: CsvParser = createParser())
     }
 
     private fun chunkLines(pageTab: String): List<TsvChunkLine> {
-        val parsedChunks = parser.parseAll(StringReader(pageTab))
-        return parsedChunks.mapIndexed(this::asTsvChunkLine)
-    }
+        val parsedChunks = parser.parse(StringReader(pageTab))
 
-    private fun asTsvChunkLine(idx: Int, values: Array<String?>): TsvChunkLine =
-        if (values.all { it.isNullOrBlank() }) TsvChunkLine(idx) else TsvChunkLine(idx, asList(values))
+        return parsedChunks.mapIndexed { idx, csvRecord ->
+            val record = csvRecord.asList()
+            when {
+                record.all(String::isBlank) -> TsvChunkLine(idx, emptyList())
+                else -> TsvChunkLine(idx, record)
+            }
+        }
+    }
 
-    private fun asList(values: Array<String?>) = values.map { it.orEmpty() }.toList()
+    private fun CSVRecord.asList(): List<String> = map { it }
 
     companion object {
-        private fun createParser() = CsvParser(createParseSettings())
-
-        /**
-         * Create CSV parser function which allow to quote special characters or multiline values as "the \n value" but
-         * at the same time allows using them explicitly like in 'the "final" version'.
-         *
-         */
-        private fun createParseSettings() = CsvParserSettings().apply {
-            format.setLineSeparator("\n")
-            format.delimiter = TAB
-            format.comment = TSV_COMMENT
-            skipEmptyLines = false
+        private fun createParser(): CSVFormat {
```

Comment:
Why are you removing these changes back? I think these are the fix for https://www.pivotaltracker.com/story/show/174642598 right?

Reply:
https://github.com/uniVocity/univocity-parsers/issues/454